src/query/procedure/mg_procedure_impl.hpp

```diff
@@ -675,8 +675,27 @@ struct mgp_vertices_iterator {
   /// @throw anything VerticesIterable may throw
   mgp_vertices_iterator(mgp_graph *graph, memgraph::utils::MemoryResource *memory)
       : memory(memory), graph(graph), vertices(graph->impl->Vertices(graph->view)), current_it(vertices.begin()) {
-    if (current_it != vertices.end()) {
+    memgraph::utils::OnScopeExit clean_up([this] { this->current_v = std::nullopt; });
+    auto *checker = graph->ctx->fine_grained_access_checker;
+
+    while (current_it != vertices.end()) {
       current_v.emplace(*current_it, graph, memory);
+
+      if (!current_v.has_value()) {
+        break;
+      }
```

Comment:
You just emplaced a value into this optional, therefore I think this check is extraneous.